Shrine/app/src/main/java/com/authentication/shrine/ui/RegisterPasswordScreen.kt

```diff
@@ -0,0 +1,315 @@
+    viewModel: RegistrationViewModel,
+    onBackClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+    credentialManagerUtils: CredentialManagerUtils,
+) {
+    val uiState = viewModel.uiState.collectAsState().value
+    var email by remember { mutableStateOf("") }
+    var password by remember { mutableStateOf("") }
+    var passwordVisible by remember { mutableStateOf(false) }
+
+    val context = LocalContext.current
+    val createRestoreKey = {
+        viewModel.createRestoreKey(
+            createRestoreKeyOnCredMan = { createRestoreCredObject ->
+                credentialManagerUtils.createRestoreKey(
+                    context = context,
+                    requestResult = createRestoreCredObject,
+                )
+            },
+        )
+    }
+
+    val onRegister = { emailAddress: String, registrationPassword: String ->
+        viewModel.onPasswordRegister(
+            username = emailAddress,
+            password = registrationPassword,
+            onSuccess = { flag ->
+                createRestoreKey()
+                navigateToHome(flag)
+            },
+            createPassword = { username: String, password: String ->
+                credentialManagerUtils.createPassword(
+                    username = username,
+                    password = password,
+                    context = context,
+                )
+            },
+            createRestoreCredential = { createRestoreCredObject ->
+                credentialManagerUtils.createRestoreKey(
+                    createRestoreCredObject,
+                    context,
+                )
+            },
+        )
+    }
+
+    RegisterPasswordScreen(
+        onRegister = onRegister,
+        uiState = uiState,
+        email = email,
+        onEmailChanged = { email = it },
+        password = password,
+        onPasswordChange = { password = it },
+        passwordVisible = passwordVisible,
+        onPasswordVisibilityToggle = { passwordVisible = !passwordVisible },
+        onBackClicked = onBackClicked,
+        modifier = modifier,
+    )
+}
+
+/**
+ * Stateless composable function that displays the registration screen.
+ *
+ * This screen allows users to create a new account by providing their email and password.
+ * It also supports signing in with passkeys.
+ *
+ * @param onRegister Callback for logging in with email and password.
+ * @param uiState The UI state of the authentication process.
+ * @param email The user's email address.
+ * @param onEmailChanged Callback for when the email address changes.
+ * @param password The user's password.
+ * @param onPasswordChange Callback for when the password changes.
+ * @param passwordVisible Whether the password is currently visible.
+ * @param onPasswordVisibilityToggle Callback for toggling the visibility of the password.
+ * @param modifier The modifier to be applied to the composable.
+ */
+@Composable
+fun RegisterPasswordScreen(
+    onRegister: (String, String) -> Unit,
```

Comment:
Can we reduce these much variables and club somehow? Looks like a long list of params here

Reply:
Sounds good, please create a ticket for refactoring and improvements.